client.go

```diff
@@ -213,7 +218,37 @@ func (client *Client) makeRPCCall(ctx context.Context, contentType string, rpcTy
 	delete(client.rpcReturnMap, corrID)
 	client.rpcReturnMutex.Unlock()
 
-	done <- &callResult
+	return &callResult
+}
+
+func (client *Client) makeRPCCall(ctx context.Context, contentType string, rpcType string, args []byte, done chan *RPCCallResult) {
```

Comment:
This needs to exit once the context has expired. Otherwise the retry will continue in a loop forever until the call is successful. It doesn't look like `tryRPC` actually uses its context, so we may just need to create another done channel, pass it in to `tryRPC` and then select between that channel and `ctx.Done()` to properly time out.